designs/2019-config-simplification/README.md

```diff
@@ -0,0 +1,627 @@
+1. Does the file pattern system actually remove the need for `--ext`?
+1. How should `files` and `ignores` be merged when a shareable config has them? Should they be overwritten or merged?
+
+## Frequently Asked Questions
+
+### Can a config be an async function?
+
+No. Right now it won't be possible to implement a config with an async function because the rest of ESLint is fully synchronous. Once we look at how to make ESLint more asynchronous, we can revisit and allow configs to be created with async functions.
+
+### Why use `exports.config` instead of `module.exports`?
+
+Using an exported key gives us more flexibility for the future if we decide that config files should be able to output more than one thing. For example, I've been thinking of a `--config-key` option that would allow users to specify which exported key should be used as their config. Users could then export multiple different keys (`config1`, `config2`, etc.) and easily switch between configs on the command line. That option is not part of this proposal because it isn't solving an existing problem and I'd rather focus on existing problems first (this proposal is already big enough).
+
+### How does this affect configuration via `package.json`?
+
+The `eslintConfig` and `eslintIgnore` keys in `package.json` will not be honored when `eslint.config.js` is found. Users could still pull that information into their `eslint.config.js` file manually if they want to.
+
+### Do shareable configs still export an object on `module.exports`?
+
+That is completely up to the shareable config. For simplicity sake, I think we still want to encourage people to do so. However, there is no longer and formal contract between ESLint and shareable configs, so developers could potentially export configs from any npm package using any exported key. They would just need to inform users about how to extend their config properly.
+
+### Is there a way to allow plugins to specify the namespace of their rules?
+
+Maybe, but even if we could do that, the namespaces aren't guaranteed to be unique.
+
+The reason we could enforce naming of both plugin packages and rule namespaces is because ESLint controlled how the plugins were loaded: users passed ESLint a string and then ESLint could both inspect the string to pull out details it needed (the plugin name without the `eslint-plugin-` prefix) and then modify the rule names to be prepended with the plugin name.
+
+In this design, the user is responsible for loading npm packages. Because we are only ever passed an object, we no longer have access to the plugin name.
+
+An earlier iteration of this design had plugins specified like this:
+
+```js
+exports.config = {
+    plugins: [
+        require("eslint-plugin-react"),
+        require("eslint-plugin-mardkown")
+    ]
+};
+```
+
+This didn't work because ESLint didn't know how to name the plugin rules (we weren't getting the plugin name, just the object). So the next iteration went to this:
```

Comment:
Can we enforce a specific format on the exported object for plugin creators, something that contains property `name` for example? It doesn't solve the issue of collision, but would not allow renaming of the plugins.

Reply:
I'm adding a new section about this. I'd like to see if it solves your concern.